go/teams/list_fast.go

```diff
@@ -39,12 +41,38 @@ func ListTeamsUnverified(ctx context.Context, g *libkb.GlobalContext, arg keybas
 	}
 
 	tracer.Stage("Server")
+
+	// We have a very simple cache in case we error out on this call. In the
+	// case of a network error we try to serve old cached data if we have some.
+	// The cache is updated on successful requests but otherwise unmaintained
+	// so should only be used if we are trying to return a best-effort result.
+	cacheKey := listTeamsUnverifiedCacheKey(meUID, queryUID, arg.IncludeImplicitTeams)
 	teams, err := getTeamsListFromServer(ctx, g, queryUID,
 		false /* all */, true /* countMembers */, arg.IncludeImplicitTeams)
-	if err != nil {
+	switch err.(type) {
+	case nil:
+		if err = g.GetKVStore().PutObj(cacheKey, nil, teams); err != nil {
+			m.CDebugf("| ListTeamsUnverified unable to put cache item: %v", err)
+		}
+	case libkb.APINetError:
+		if found, cerr := g.GetKVStore().GetInto(&teams, cacheKey); cerr != nil || !found {
+			// Nothing we can do here.
+			m.CDebugf("| ListTeamsUnverified unable to get cache item: %v, found: %v", cerr, found)
+			return nil, err
+		}
+	default:
 		return nil, err
 	}
 
+	res := &keybase1.AnnotatedTeamList{
```

Comment:
it seemed better to short circuit here so we don't do the UPAK load for nothing: https://github.com/keybase/client/pull/13291/files#diff-c208ef40eba88b78a63656fe1dbb6ecfR81